src/axom/core/CMakeLists.txt

```diff
@@ -107,6 +107,12 @@ blt_add_library( NAME        core
                  OBJECT      TRUE
                  )
 
+# Add the AXOM_DEBUG compile definition, if non-empty
```

Comment:
There are lots of small changes, so I'll point out some of the key ones.
This attaches `AXOM_DEBUG` to core, when defined